File: app/src/main/java/knf/kuma/changelog/ChangeAdapter.java
package knf.kuma.changelog;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.ButterKnife;
import knf.kuma.R;
import knf.kuma.changelog.objects.Change;
import knf.kuma.changelog.objects.Release;

public class ChangeAdapter extends RecyclerView.Adapter<ChangeAdapter.ChangeItem> {

    private List<Change> changes = new ArrayList<>();

    public ChangeAdapter(Release release) {
        this.changes = release.changes;
    }

    @Override
    public ChangeItem onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ChangeItem(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_release, parent, false));
    }

    @Override
    public void onBindViewHolder(ChangeItem holder, int position) {
        Change change = changes.get(position);
        setType(holder.type, change.type);
        holder.description.setText(change.text);
    }

    private void setType(TextView textView, String type) {
        switch (type) {
            case "new":
                textView.setText("Nuevo");
                textView.setBackgroundResource(R.drawable.chip_new);
                break;
            default:
            case "change":
                textView.setText("Cambio");
                textView.setBackgroundResource(R.drawable.chip_change);
                break;
            case "fix":
                textView.setText("Arreglo");
                textView.setBackgroundResource(R.drawable.chip_error);
                break;
        }
    }

    @Override
    public int getItemCount() {
        return changes.size();
    }

    class ChangeItem extends RecyclerView.ViewHolder {
        @BindView(R.id.type)
        TextView type;
        @BindView(R.id.description)
        TextView description;

        public ChangeItem(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/knf/kuma/changelog/ChangeAdapter.kt
package knf.kuma.changelog

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import knf.kuma.R
import knf.kuma.changelog.objects.Release
import kotlinx.android.synthetic.main.item_release.view.*

internal class ChangeAdapter(release: Release) : RecyclerView.Adapter<ChangeAdapter.ChangeItem>() {

    private val changes = release.changes

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ChangeItem {
        return ChangeItem(LayoutInflater.from(parent.context).inflate(R.layout.item_release, parent, false))
    }

    override fun onBindViewHolder(holder: ChangeItem, position: Int) {
        val change = changes[position]
        setType(holder.type, change.type)
        holder.description.text = change.text
    }

    @SuppressLint("SetTextI18n")
    private fun setType(textView: TextView?, type: String) {
        when (type) {
            "new" -> {
                textView!!.text = "Nuevo"
                textView.setBackgroundResource(R.drawable.chip_new)
            }
            "change" -> {
                textView!!.text = "Cambio"
                textView.setBackgroundResource(R.drawable.chip_change)
            }
            "fix" -> {
                textView!!.text = "Arreglo"
                textView.setBackgroundResource(R.drawable.chip_error)
            }
            else -> {
                textView!!.text = "Cambio"
                textView.setBackgroundResource(R.drawable.chip_change)
            }
        }
    }

    override fun getItemCount(): Int {
        return changes.size
    }

    internal inner class ChangeItem(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var type: TextView = itemView.type
        var description: TextView = itemView.description
    }
}
